reconcile/utils/jinja2/utils.py

```diff
@@ -188,6 +193,118 @@ def list_s3_objects(
         )
 
 
+def _process_sloth_output(output_file_path: str) -> str:
+    """Process sloth output to make it compliant with the prometheus-rule-1 schema"""
+    with open(output_file_path, encoding="utf-8") as f:
+        content = f.read()
+        # Strip leading whitespace and document separator since we're embedding in existing YAML
+        content = content.lstrip()
+        content = content.removeprefix("---\n")
+        # Remove sloth_* labels and title annotation
+        content = re.sub(r"^(\s+)sloth_\w+:.*\n", "", content, flags=re.MULTILINE)
+        content = re.sub(
+            r"^(\s+)title:.*\n(?:(?:\1\s+.*\n)*)", "", content, flags=re.MULTILINE
+        )
+        # Remove labels sections from recording rules
+        # sloth adds numerous labels that violate schema and we don't have any functionality based off recording rule labels
+        content = re.sub(
+            r"^(\s+)labels:\s*\n(?=\s*(?:- (?:record|alert|name):|$))",
+            "",
+            content,
+            flags=re.MULTILINE,
+        )
```

Comment:
please parse output into structured model class instead of change text via regular expression

Reply:
addressed within https://github.com/app-sre/qontract-reconcile/pull/5153/commits/25db123c4e713facdcbbfad218bef8b2203d1e9d